lib/reek/code_comment.rb

```diff
@@ -20,19 +21,42 @@ class CodeComment
     DISABLE_DETECTOR_CONFIGURATION = '{ enabled: false }'.freeze
     MINIMUM_CONTENT_LENGTH         = 2
     LEGACY_SEPARATOR               = ':'.freeze
+    UNKNOWN_SMELL_DETECTOR_MESSAGE = <<-EOS.freeze
+
+      Error: You are trying to configure an unknown smell detector '%s' in one
+      of your source code comments.
+      The source is '%s' and the comment belongs to the expression starting in line %d.
+      Here's the original comment:
+
+      %s
+
+      Please see the Reek docs for:
+        * how to configure Reek via source code comments: https://github.com/troessner/reek/blob/master/docs/Smell-Suppression.md
+        * what smell detectors are available: https://github.com/troessner/reek/blob/master/docs/Code-Smells.md
+      Update the offensive comment (or remove it if no longer applicable) and re-run Reek.
+
+    EOS
 
     attr_reader :config
 
     #
     # @param comment [String] - the original comment as found in the source code
-    #   E.g.:
-    #   "\n        # :reek:Duplication: { enabled: false }\n      "
+    # @param line [Integer] - start of the expression the comment belongs to
+    # @param source [String] - Path to source file or "string"
     #
-    def initialize(comment)
+    def initialize(comment:, line:, source:)
       @original_comment  = comment
       @config            = Hash.new { |hash, key| hash[key] = {} }
 
       @original_comment.scan(CONFIGURATION_REGEX) do |detector, _option_string, options|
+        unless Smells::SmellDetector.valid_detector?(detector)
```

Comment:
> But then we would separate creation from validation.

Yes, this would be an external validation. In general, I tend to avoid having internal validations if it creates too many dependencies.

However, it's probably best to go with this implementation for now, and then I can see if I can refactor it later and we can discuss this more then.